tools/cli/adminCommands.go

```diff
@@ -167,19 +150,83 @@ func AdminDescribeWorkflow(c *cli.Context) {
+	adminClient := cFactory.ServerAdminClient(c)
+
+	domain := getRequiredGlobalOption(c, FlagDomain)
+	wid := getRequiredOption(c, FlagWorkflowID)
+	rid := c.String(FlagRunID)
+
+	ctx, cancel := newContext(c)
+	defer cancel()
+
+	resp, err := adminClient.DescribeWorkflowExecution(ctx, &admin.DescribeWorkflowExecutionRequest{
+		Domain: common.StringPtr(domain),
+		Execution: &shared.WorkflowExecution{
+			WorkflowId: common.StringPtr(wid),
+			RunId:      common.StringPtr(rid),
+		},
+	})
+	if err != nil {
+		ErrorAndExit("Get workflow mutableState failed", err)
+	}
+	return resp
+}
+
 // AdminDeleteWorkflow describe a new workflow execution for admin
 func AdminDeleteWorkflow(c *cli.Context) {
-	domainID := getRequiredOption(c, FlagDomainID)
 	wid := getRequiredOption(c, FlagWorkflowID)
-	rid := getRequiredOption(c, FlagRunID)
-	if !c.IsSet(FlagShardID) {
-		ErrorAndExit("shardID is required", nil)
-	}
-	shardID := c.Int(FlagShardID)
+	rid := c.String(FlagRunID)
 
+	resp := describeMutableState(c)
+	shardID := resp.GetShardId()
+	msStr := resp.GetMutableStateInDatabase()
+	ms := persistence.WorkflowMutableState{}
+	err := json.Unmarshal([]byte(msStr), &ms)
+	if err != nil {
+		ErrorAndExit("json.Unmarshal err", err)
+	}
+	domainID := ms.ExecutionInfo.DomainID
+	skipError := c.Bool(FlagSkipErrorMode)
 	session := connectToCassandra(c)
+	if ms.ExecutionInfo.EventStoreVersion == persistence.EventStoreVersionV2 {
+		branchInfo := shared.HistoryBranch{}
+		thriftrwEncoder := codec.NewThriftRWEncoder()
+		err := thriftrwEncoder.Decode(ms.ExecutionInfo.BranchToken, &branchInfo)
+		if err != nil {
+			ErrorAndExit("thriftrwEncoder.Decode err", err)
+		}
+		fmt.Println("deleting history events for ...")
+		prettyPrintJSONObject(branchInfo)
+		histV2 := cassp.NewHistoryV2PersistenceFromSession(session, bark.NewNopLogger())
+		err = histV2.DeleteHistoryBranch(&persistence.InternalDeleteHistoryBranchRequest{
+			BranchInfo: branchInfo,
+		})
+		if err != nil {
+			if skipError {
+				fmt.Println("failed to delete history, ", err)
+			} else {
+				ErrorAndExit("DeleteHistoryBranch err", err)
+			}
+		}
+	} else {
+		histV1 := cassp.NewHistoryPersistenceFromSession(session, bark.NewNopLogger())
+		err = histV1.DeleteWorkflowExecutionHistory(&persistence.DeleteWorkflowExecutionHistoryRequest{
+			DomainID: domainID,
+			Execution: shared.WorkflowExecution{
+				WorkflowId: common.StringPtr(wid),
+				RunId:      common.StringPtr(rid),
+			},
+		})
+		if err != nil {
+			if skipError {
+				fmt.Println("failed to delete history, ", err)
+			} else {
+				ErrorAndExit("DeleteWorkflowExecutionHistory err", err)
+			}
+		}
+	}
 
-	var err error
 	permanentRunID := "30000000-0000-f000-f000-000000000001"
 	selectTmpl := "select execution from executions where shard_id = ? and type = 1 and domain_id = ? and workflow_id = ? and run_id = ? "
 	deleteTmpl := "delete from executions where shard_id = ? and type = 1 and domain_id = ? and workflow_id = ? and run_id = ? "
```

Comment:
when deleting the current execution record, plz use 'if current_run_id = ?' condition check, in case of racing condition